wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1018,12 +1019,29 @@ instance BuildableSafeGen PasswordUpdate where
         pwdOld
         pwdNew
 
+data Accuracy
+    = Accurate
+    | LowerBound
+    deriving (Eq, Show, Generic)
+
+instance Arbitrary Accuracy where
+    arbitrary = elements [Accurate, LowerBound]
+
+instance ToSchema Accuracy where
+    declareNamedSchema = genericDeclareNamedSchema defaultSchemaOptions
+
+deriveJSON Serokell.defaultOptions ''Accuracy
 
 -- | 'EstimatedFees' represents the fees which would be generated
 -- for a 'Payment' in case the latter would actually be performed.
-data EstimatedFees = EstimatedFees {
-    feeEstimatedAmount :: !(V1 Core.Coin)
-  } deriving (Show, Eq, Generic)
+data EstimatedFees = EstimatedFees
+    { feeEstimatedAmount :: !(V1 Core.Coin)
+    -- ^ The estimated amount.
+    , feeAccuracy :: !Accuracy
+    -- ^ The accuracy of this estimate. If an account has insufficient
+    -- funds to perform a transaction, then we return 'LowerBound', as
+    -- adding additional inputs to a transaction will increase the fee.
```

Comment:
I know that I talked with @adinapoli-iohk about keeping the error response, but this ended up being an easier way to go. It also preserves the "successful" semantics vs an error condition.

We say *estimated* fees, which is why I am okay with this change.

If we decide to go and do an HTTP error, then I can change this.